File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    
    for num in arr:
        if len(answer) == 0:
            answer.append(num)
        
        if answer[-1] == num:
            continue
        else:
            answer.append(num)
    
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <bits/stdc++.h>

using namespace std;

stack<int> st;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;

    for (int n : arr) {
        if (st.empty()) st.push(n);
        else if (st.top() != n) st.push(n); 
    }
    
    while (st.size()) {
        answer.push_back(st.top());
        st.pop();
    }
    
    reverse(answer.begin(), answer.end());

    return answer;
}